File: JavaScript/uri-1046.js
// URI - 1046: https://www.urionlinejudge.com.br/judge/pt/problems/view/1046

const input = require('fs').readFileSync('/dev/stdin', 'utf-8');
const values = input.split(' ').map(value => parseInt(value));

console.log(calculateGameDuration(values[0], values[1]));

function calculateGameDuration(startTime, endTime) {
  if (startTime > endTime) 
    return `O JOGO DUROU ${(endTime + 24) - startTime} HORA(S)`;
  else if (startTime === endTime)
    return 'O JOGO DUROU 24 HORA(S)';
  else 
    return `O JOGO DUROU ${endTime - startTime} HORA(S)`;
}

File: URI/Ruby/uri-1046.rb
# frozen_string_literal: true

# URI - 1046: https://www.urionlinejudge.com.br/judge/pt/problems/view/1046

def calculate_game_duration(start_time, end_time)
  return 'O JOGO DUROU 24 HORA(S)' if start_time == end_time

  duration = if start_time < end_time
               end_time - start_time
             else
               (end_time + 24) - start_time
             end
  "O JOGO DUROU #{duration} HORA(S)"
end

input = gets.split(' ').map(&:to_i)
puts calculate_game_duration(input[0], input[1])
